src/main/scala/io/iohk/ethereum/domain/Blockchain.scala

```diff
@@ -235,10 +236,11 @@ class BlockchainImpl(
     }
 
   override def getAccountStorageAt(rootHash: ByteString, position: BigInt): ByteString = {
-    storageMpt(
+    val value = storageMpt(
       rootHash,
       nodesKeyValueStorageFor(None)
-    ).get(UInt256(position)).getOrElse(UInt256(0)).bytes
+    ).get(position).getOrElse(BigInt(0))
+    ByteString(BigIntegers.asUnsignedByteArray(value.bigInteger))
```

Comment:
Hmm, is this different from `ByteString(value.toByteArray)` ?

Reply:
seems it's not different

Reply:
I think it is, compare:

```
scala> BigIntegers.fromUnsignedByteArray(BigIntegers.asUnsignedByteArray(BigInt(-1).bigInteger))
res3: java.math.BigInteger = 255

scala> BigInt(BigInt(-1).toByteArray)
res4: scala.math.BigInt = -1
```

I think the latter method should be used in here and in `RLPImplicits`

Reply:
`ByteString(BigIntegers.asUnsignedByteArray(value.bigInteger))` and `ByteString(value.toByteArray)` seem to work the same way, it's the other way conversion that behaves differently

Reply:
`BigIntegers.fromUnsignedByteArray` vs BigInt apply

Reply:
Changed to `ByteString(value.toByteArray)`

Reply:
> I think the latter method should be used in here and in RLPImplicits
we can't just use BigInt apply in RLPImplicits, we need to decode the bytes from rlp format